app/workers/share_worker.rb

```diff
@@ -9,8 +9,7 @@ class ShareWorker
   # @param answer_id [Integer] the user id
   # @param service [String] the service to post to
   def perform(user_id, answer_id, service)
-    service_type = "Services::#{service.camelize}"
-    user_service = User.find(user_id).services.find_by(type: service_type)
+    user_service = find_service(user_id, service)
```

Comment:
could use an instance variable here

```suggestion
    @user_service = User.find(user_id).services.find_by(type: "Services::#{service.camelize}")
```